From 92a8ff08546ba904e50e2a83ed81f928f6d36236 Mon Sep 17 00:00:00 2001
From: Eberhard Beilharz <eb1@sil.org>
Date: Fri, 27 Apr 2012 14:59:53 +0200
Subject: [PATCH 2/2] Xamarin-4690: Fix MoveToRoot()

XNodeNavigator.MoveToRoot() needs to move to the root
element even when node.Document is not set.

Change-Id: I188bf9fa6c6cae1ffebfa006b518301508a891bb
---
 .../System.Xml.Linq/XNodeNavigator.cs              |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

Index: fieldworks-mono/mcs/class/System.Xml.Linq/System.Xml.Linq/XNodeNavigator.cs
===================================================================
--- fieldworks-mono.orig/mcs/class/System.Xml.Linq/System.Xml.Linq/XNodeNavigator.cs
+++ fieldworks-mono/mcs/class/System.Xml.Linq/System.Xml.Linq/XNodeNavigator.cs
@@ -413,8 +413,14 @@
 
 		public override void MoveToRoot ()
 		{
-			node = node.Document ?? node;
 			attr = null;
+			if (node.Document != null) {
+				node = node.Document;
+			}
+			else {
+				while (node.Owner != null)
+					node = node.Owner;
+			}
 		}
 	}
 }
